Added typedefinitions for the npm package jsonld (jsonld.js)

jason076 opened the pull request:
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

rbuckton commented on types/jsonld/index.d.ts:
```diff
@@ -0,0 +1,217 @@
+    type Flatten = Common;
+
+    interface Frame {
+        embed?: '@last'| '@always'|'@never'|'@link';
```
Spacing is inconsistent.

rbuckton commented on types/jsonld/index.d.ts:
```diff
@@ -0,0 +1,217 @@
+export function toRDF(input: Document, options: Options.ToRdf, callback: Callback<RdfDataSet>): void;
+export function toRDF(input: Document, options?: Options.ToRdf): Promise<RdfDataSet>;
+
+export let JsonLdProcessor: JsonLdProcessor;
```
This is incorrect. What is exported is not a `JsonLdProcessor`-typed value, but a `JsonLdProcessor` class: https://github.com/digitalbazaar/jsonld.js/blob/master/lib/JsonLdProcessor.js#L7

jason076 replied:
The class is just having the 3 static functions "expand", "flatten" and "compact". If I use a class instead of an interface the linter is complaining about that the class has only static properties. In my opinion it is correct to use the interface here. I have added an according section to the test file, to test if it is working.

jason076 replied:
Should I change something anyway or can I mark it as resolved?

rbuckton commented on types/jsonld/index.d.ts:
```diff
@@ -0,0 +1,217 @@
+ * The interfaces are usefull to avoid code replication.
+ */
+
+declare namespace Options {
```
If these `Options` types are not also exported, consumers will have a hard time using the options types themselves if they want to reuse data. It also makes it impossible to modify the options types via module augmentations if a consumer finds something is missing and wants to test the change locally.

jason076 replied:
You were right. I have exported them.

rbuckton commented on types/jsonld/jsonld-spec.d.ts:
```diff
@@ -0,0 +1,40 @@
+export type JsonLd = JsonLdObj|JsonLdArray;
+
+type DOMString = string;
+type LoadDocumentCallback = Promise <Url>;
```
Remove space between Identifier and Type Argument list

jason076 replied:
Fixed

DanielRosenwasser commented:
Just as a heads up, your commits don't seem to be associated with your GitHub account. While this isn't technically a problem, you might care if you want more appropriate attribution. You can either make sure your GitHub account is associated with the email address you're using for your commits, or rebase and amend your commits to fix the author name and email.

jason076 commented:
> Just as a heads up, your commits don't seem to be associated with your GitHub account. While this isn't technically a problem, you might care if you want more appropriate attribution. You can either make sure your GitHub account is associated with the email address you're using for your commits, or rebase and amend your commits to fix the author name and email.

Thanks for the hint. I am new to github. 
I have rebased the commits onto the current master branch and changed the email address. They should be associated with my GitHub account now

DanielRosenwasser commented:
Thanks!
